packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -320,10 +320,11 @@ abstract class FlutterCommand extends Command<void> {
   }
 
   BuildInfo getBuildInfo() {
-    TargetPlatform targetPlatform;
-    if (argParser.options.containsKey('target-platform') &&
-        argResults['target-platform'] != 'default') {
-      targetPlatform = getTargetPlatformForName(argResults['target-platform']);
+    Iterable<TargetPlatform> targetPlatforms;
+    if (argParser.options.containsKey('target-platform')) {
+      targetPlatforms = argResults['target-platform'].map<TargetPlatform>(getTargetPlatformForName);
```

Comment:
`build_aot.dart` and `build_bundle.dart` both still specify a single `--target-platform` argument, which if it hits this code path will cause a runtime error trying to call `map()` on `String`.

Reply:
This isn't an issue anymore since I refactored the code, and `BuildInfo` doesn't have target platform anymore.